openfe/setup/ligand_network_planning.py

```diff
@@ -82,67 +99,57 @@ def generate_radial_network(
     """
     if isinstance(mappers, AtomMapper):
         mappers = [mappers]
-    mappers = [_hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper)
-               else m for m in mappers]
+    mappers = [
+        _hasten_lomap(m, ligands) if isinstance(m, LomapAtomMapper) else m
+        for m in mappers
+    ]
 
     # handle central_ligand arg possibilities
     # after this, central_ligand is resolved to a SmallMoleculeComponent
     if isinstance(central_ligand, int):
         ligands = list(ligands)
         try:
             central_ligand = ligands[central_ligand]
+            ligands.remove(central_ligand)
         except IndexError:
-            raise ValueError(f"index '{central_ligand}' out of bounds, there are "
-                             f"{len(ligands)} ligands")
+            raise ValueError(
```

Comment:
I realize this is true of the original code as well, but why are we replacing an `IndexError` with a `ValueError`? Don't we just want an `IndexError` with a custom error message?

Reply:
this is a fair question, where I don't know the answer to :(

Reply:
I believe the intent is that it's a `ValueError` because the "incorrect value" (i.e. `central_ligand`) was passed.

That being said, I'm pretty sure we just put a ValueError everywhere because it's easier than really thinking about our error types.